tests/new_api_find_multiple_threads.rs

```diff
@@ -0,0 +1,31 @@
+use librpm::{Db, Index};
+
+use std::path::Path;
+use std::sync::{Arc, Mutex};
+use std::thread;
+
+#[test]
+fn db_find_test_multiple() {
+    let db = Arc::new(Mutex::new(Db::open::<&Path>().unwrap()));
+
+    {
+        let db = db.clone();
+        thread::spawn(move || {
+            let mut matches = Index::Name.find(&db.lock().unwrap(), "glibc-common");
```

Comment:
I got an idea.

The problem might be that `MutexGuard` produced by `&db.lock().unwrap()` dies inside of expression while calculating the function argument, rather than living up until end of the block.

I'll try to check that.

Reply:
So yes, it was an issue from Rust point of view, as lifetime of `Iter` wasn't tied to lifetime of `Db` and `MutexGuard` could be dropped earlier.
But it doesn't prevent the actual BerkeleyDB "thread died in DB" problem.